dispatch/FileCache.h

```diff
@@ -333,23 +355,33 @@ class FileCache {
 
     /**
      * @brief Initialize the cache.
-     * @param cache_dir Directory on some filesystem where the cache will be stored.
+     * @param cache_dir Directory on some filesystem where the cache will be stored. This
+     * directory is made if it does not exist.
      * @param size Allow this many bytes in the cache
      * @param target_size When purging, remove items until this many bytes remain.
      * @return False if the cache object could not be initialized, true otherwise.
      */
     virtual bool initialize(const std::string &cache_dir, long long size, long long purge_size) {
```

Comment:
Why a signed int for size and purge size? Why not `uint64_t` ??

Reply:
I made these sizes signed because they are used in comparisons where a negative value is possible and casting would cause that to 'flip' over to a huge positive value (so the cache would never be purged, etc.). I figured that was OK since signed 64 bit int is still really large.